text/0000-wallet-update-process.md

```diff
@@ -0,0 +1,170 @@
+
+* `start_update_thread(interval)` => Starts the update thread, calling the Overall Update Process at the frequency specified
+* `stop_update_thread()` => Stops the update thread
+* `get_update_status()` => Returns the current status of the update thread, along with as much information as possible about what the thread is doing and, in the case of the `scan` process, the percentage complete.
+
+Further, if the update thread is currently running, commands that previously always called the Overall Update Process (such as `info`, `txs`) etc do not call the update process. This means that when invoked via the command line, the behaviour is unchanged. However, wallets running the Owner API can instead choose to run and monitor the background update thread at the desired frequency while keeping the user informed of any long-running update processes.
+
+### Transaction TTL
+
+The Update process outlined above does not cover the case where a slate is exchanged and outputs are locked, but one party doesn't complete the transaction for whatever reason. 
+
+Previously, a situation such as this meant that the outputs associated with a transaction sat locked indefinitely within a user's wallet, with the only way to unlock them a manual `cancel` command (or a `check-repair` with the `delete-unconfirmed` flag set). Worse, the wallet's default selection strategy was set to `all` meaning an entire wallet's contents usually became locked on each transaction, and thus a single uncompleted transaction would render a user's entire wallet balance unspendable.
+
+To rectify these problems, the following changes are made:
+
+* A Time-To-Live (`TTL`) field is added to the Slate, which is defined as the last block height at which a wallet should attempt to complete a transaction. This field should be respected at the wallet-exchange level, but is not currently commit to at a consensus level. If a wallet detects a particlar transaction's TTL has expired, it will automatically cancel the transaction and unlock associated outputs. Note this doesn't prevent the other party sending the transaction to the node after this happens, but there is no guarantee the outputs will still be available. However, if this does happen the `scan` command will correct the wallet state.
+* The default output selection method is changed to `smallest`, to prevent all wallet amounts being locked on every transaction.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Do we limit the Kernel Lookup part of the process to just cases where there is no change output. Is there any benefit to calling it for all outstanding transactions on each lookup (or any particular downside to doing so?)
+- Can we add the TTL field at consensus level? Making it consensus means much less scope for inconsistency between wallets overall, but how would it be commit to? (Could it be inclded in the kernel excess somehow, for example?)
```

Comment:
The only thing we could do is add it to the kernel and become part of the signature, but it could end up being really bad for privacy. For example, if Grin++ uses a different algo for calculating TTL than grin wallet, allowing you to quickly identify grin wallet kernels and Grin++ kernels, effectively splitting the anon set in half, I would recommend not supporting this at the consensus level for this reason.